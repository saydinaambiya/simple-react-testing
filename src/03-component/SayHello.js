import * as React from 'react';

const SayHello = props => {
    const { person } = props;
    return (
        <>
            <h1>Hello {person}!</h1>
            <p>Welcome Home</p>
        </>
    );
};

export default SayHello;