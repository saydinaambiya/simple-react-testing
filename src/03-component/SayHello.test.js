import * as React from "react";
import {render, screen} from "@testing-library/react";
import SayHello from "./SayHello";

describe("SayHello component", () => {
    test("should display the person name", () => {
        const person = "John";

        render(<SayHello person={person}/>)
        const element = screen.getByText("Hello John!");

        expect(element.tagName).toEqual("H1");
    })

    test("should display say hello component correctly", () => {
        const person = "John";

        const {asFragment} = render(<SayHello person={person}/>)

        expect(asFragment()).toMatchSnapshot();
    })
})