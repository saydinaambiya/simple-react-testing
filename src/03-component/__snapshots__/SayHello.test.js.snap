// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SayHello component should display say hello component correctly 1`] = `
<DocumentFragment>
  <h1>
    Hello John!
  </h1>
  <p>
    Welcome Home
  </p>
</DocumentFragment>
`;
